### PR TITLE
build options docs

### PR DESCRIPTION
**Pull request by nzoschke**
## Release Playbook
- [ ] Rebase against master
- [ ] Code review
- [ ] Merge into master
- [ ] Verify changes at http://site-staging.convox.com
- [ ] Promote release on `site-production` in the `convox/production` Rack

### CODE REVIEW

File: _management/advanced-build-options.md [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+
+### Build Artifact Caching
+
+App language dependency management tools like [Ruby's Bundler](http://bundler.io/) and [Python PIP](https://pypi.python.org/pypi/pip) take care to reuse packages that have already been built and installed.
+
+Docker builds, however, are stateless and implement their own [image caching](https://docs.docker.com/engine/userguide/eng-image/dockerfile_best-practices/#/build-cache) which ends up having to rebuild all dependencies on any `Gemfile` change.
+
+Convox offers a convention to get the optimal dependency management behavior.
+
+After a `convox build`, any files in `/var/cache/build` are extracted and saved to an archive in S3. At the beginning of the next `convox build`, this archive is downloaded and extracted to a `.cache/` directory in the app source code.
```
Comment (mwarkentin) [inline _management/advanced-build-options.md]: Is there a way to "uncache" something?

Comment (nzoschke) [inline _management/advanced-build-options.md]: Locally you can delete the directories. On a rack I'm not sure yet. It seems like `--no-cache` should not use this cache too.

File: _management/advanced-build-options.md [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+
+You can then instruct Docker to use the artifacts from the last build and update them for the next build by:
+
+```Dockerfile
+COPY Gemfile      /app/Gemfile
+COPY Gemfile.lock /app/Gemfile.lock
+COPY .cache/build/gems /var/cache/build/gems
```
Comment (nzoschke) [inline _management/advanced-build-options.md]: In https://github.com/convox/rack/pull/1909 this has changed to:

`COPY .cache/build/buildpack /var/cache/build/buildpack`

Comment (MiguelMoll) [inline _management/advanced-build-options.md]: Believe this is context specific. If you are running build cache with a ruby app, you'd use `COPY .cache/build/gems /var/cache/build/gems` vs `COPY .cache/build/buildpack /var/cache/build/buildpack` when working with a buildpack

**Pull request merged**